src/components/labelPrinting/LabelPrintingForm.vue

```diff
@@ -112,114 +95,74 @@
-              listSuffixedBarcodes,
-              noOfLabels,
-            )
-            break
-          case this.suffix() && !applyLabels:
-            //Add only suffix to the barcodes to be printed
-            listSuffixedBarcodes = this.appendSuffix(barcodes, listSuffixedBarcodes)
-            break
-          case applyLabels && !this.suffix():
-            //Add only label number to the barcodes to be printed
-            listSuffixedBarcodes = this.appendLabels(barcodes, listSuffixedBarcodes, noOfLabels)
-            break
-          case !applyLabels && !this.suffix():
-            //Just the plain barcodes list to be printed
-            for (let barcode of barcodes) {
-              listSuffixedBarcodes.push(barcode)
-            }
-            break
-        }
-      }
-      return listSuffixedBarcodes
     },
-    appendSuffix(barcodes, listSuffixedBarcodes) {
-      for (let barcode of barcodes) {
-        listSuffixedBarcodes.push(barcode.concat('-', this.suffix()))
-      }
-      return listSuffixedBarcodes
+    suffixOptions() {
+      return createSuffixDropdownOptions(SuffixList)
     },
-    appendLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
+    suffixItems() {
+      return createSuffixItems(SuffixList)
     },
-    appendSuffixWithLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', this.suffix(), '-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
-    },
-    suffix() {
-      return this.form.selectedSuffix === 'No suffix' ? '' : this.form.selectedSuffix.slice(0, 4)
-    },
-    printerName() {
-      return this.form.selectedPrinterName
+    labels() {
+      const date = getCurrentDate()
+      const suffixItem = this.suffixItems[this.form.suffix]
+
+      // it is possible for there to be no barcodes so we need to add a guard
+      const splitSourceBarcodeList = this.form.sourceBarcodeList?.split(/\r?\n|\r|\n/g) || []
+
+      return createLabelsFromBarcodes({
+        sourceBarcodeList: splitSourceBarcodeList,
+        date,
+        suffixItem,
+        numberOfLabels: this.form.numberOfLabels,
+      })
     },
-    onSubmit(event) {
+  },
+  methods: {
+    /*
+      Creates the print job and shows a success or failure alert
+      @param {event}
+    */
+    async printLabels(event) {
       event.preventDefault()
-      this.sendPrintRequest()
+
+      const { success, message = {} } = await this.createPrintJob({
+        printerName: this.form.printerName,
+        labels: this.labels,
+        copies: this.form.copies,
+      })
+
+      this.showAlert(message, success ? 'success' : 'danger')
+
+      return { success, message }
```

Comment:
Does this need to be returned? If it's only added for the test maybe test the `showAlert` instead

Reply:
Again I did think about that. My view was that it is a function so returning something should be standard and it felt more natural to test the return rather than something that was independent. Not sure either way.